packages/transaction-controller/src/utils.ts

```diff
@@ -207,3 +208,23 @@ export function getAndFormatTransactionsForNonceTracker(
       };
     });
 }
+
+/**
+ * Checks whether a given transaction matches the specified network or chain ID.
+ * This function is used to determine if a transaction is relevant to the current network or chain.
+ *
+ * @param transaction - The transaction metadata to check.
+ * @param chainId - The chain ID of the current network.
+ * @param networkId - The network ID of the current network.
+ * @returns A boolean value indicating whether the transaction matches the current network or chain ID.
+ */
+export function transactionMatchesNetwork(
+  transaction: TransactionMeta,
+  chainId: Hex,
+  networkId: string | null,
+) {
+  if (typeof transaction.chainId !== 'undefined') {
+    return transaction.chainId === chainId;
+  }
+  return transaction.networkID === networkId;
```

Comment:
If `networkId` can be `null` as per the type, do we also need a truthy check here as I assume we don't want a match if both values are `null` or `undefined`?

Reply:
Good catch I've included an extra truthy check.